aesara/tensor/random/type.py

```diff
@@ -14,26 +18,49 @@ class RandomStateType(Type):
     with the `==` operator.  This `Type` exists to provide an equals function
     that is used by `DebugMode`.
 
+    Also works with a `dict` derived from RandomState.get_state() unless
+    the `strict` argument is explicitly set to `True`.
+
     """
 
     def __repr__(self):
         return "RandomStateType"
 
     @classmethod
     def filter(cls, data, strict=False, allow_downcast=None):
-        if cls.is_valid_value(data):
+        if cls.is_valid_value(data, strict):
             return data
         else:
             raise TypeError()
 
     @staticmethod
-    def is_valid_value(a):
-        return isinstance(a, np.random.RandomState)
+    def is_valid_value(a, strict):
+        if isinstance(a, np.random.RandomState):
+            return True
+
+        if not strict and isinstance(a, dict):
+            gen_keys = ["bit_generator", "gauss", "has_gauss", "state"]
+            state_keys = ["key", "pos"]
+
+            for key in gen_keys:
+                if key in a:
+                    continue
+                return False
+
+            for key in state_keys:
+                if key in a["state"]:
+                    continue
+                return False
```

Comment:
Looks like we could use some simple tests for these conditions in `tests/tensor/random/test_type.py`.